runtime/gc_base/HotFieldUtil.cpp

```diff
@@ -0,0 +1,163 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "j9.h"
+#include "j9cfg.h"
+
+#if defined(J9VM_GC_MODRON_SCAVENGER) || defined(J9VM_GC_VLHGC)
+
+#include "HotFieldUtil.hpp"
+#include "GCExtensions.hpp"
+
+/* Value used to help with the incrementing of the gc count between hot field sorting for dynamicBreadthFirstScanOrdering */
+#define INCREMENT_GC_COUNT_BETWEEN_HOT_FIELD_SORT 100
+
+/* Minimum hotness value for a third hot field offset if depthCopyThreePaths is enabled for dynamicBreadthFirstScanOrdering */
+#define MINIMUM_THIRD_HOT_FIELD_HOTNESS 50000
+
+void
+MM_HotFieldUtil::sortAllHotFieldData(J9JavaVM *javaVM, MM_GCExtensions *extensions)
+{	
```

Comment:
It's enough to just pass Ext - you can get JavaVM from there.

Reply:
I think would be better to pass JavaVM and get GCExtensions from it

Reply:
i'm fine with that, too.

Reply:
I looked to implementations for both functions and there is no difference really, so I am fine with either way...

Reply:
Ok 👍  I made the change to pass Extensions and then changed it to pass JavaVM. If both are fine then I will just pass JavaVM.

Reply:
pass just JavaVM (here and then to be consistent in the other methods, too)